tests/testthat/test-deploymentTarget.R

```diff
@@ -0,0 +1,159 @@
+    accountInfo = fakeAccountInfo(ron = list(
+      username = "foo",
+      server = "test"
+    ))
+  )
+
+  app_dir <- withr::local_tempdir()
+  file.create(file.path(app_dir, "app.R"))
+
+  target <- deploymentTarget(
+    app_dir,
+    appName = "test",
+    appTitle = "mytitle",
+    appId = "123",
+    account = "ron"
+  )
+  expect_equal(target$appId, "123")
+  expect_equal(target$username, "foo")
+})
+
+test_that("errors if no previous deployments and multiple accounts", {
+  mockr::local_mock(accounts = fakeAccounts("ron", c("foo1", "foo2")))
+
+  app_dir <- withr::local_tempdir()
+  file.create(file.path(app_dir, "app.R"))
+
+  expect_snapshot(error = TRUE, {
+    deploymentTarget(app_dir)
+    deploymentTarget(app_dir, appName = "test")
+  })
+})
+
+test_that("handles accounts if only server specified", {
+  mockr::local_mock(
+    accounts = fakeAccounts(c("ron", "john"), "foo"),
+    accountInfo = fakeAccountInfo(ron = list(
+      username = "foo",
+      server = "test"
+    ))
+  )
+
+  app_dir <- withr::local_tempdir()
+  file.create(file.path(app_dir, "app.R"))
+
+  expect_snapshot(deploymentTarget(app_dir, server = "foo"), error = TRUE)
+
+  target <- deploymentTarget(app_dir, server = "foo", account = "ron", appId = "123")
+  expect_equal(target$username, "foo")
+})
+
+test_that("errors if multiple deployments", {
+  mockr::local_mock(accounts = fakeAccounts("ron", c("foo1", "foo2")))
+
+  app_dir <- withr::local_tempdir()
+  saveDeployment(
+    app_dir,
+    name = "test",
+    title = "",
+    username = "ron",
+    account = "ron",
+    server = "foo1",
+    hostUrl = "",
+    appId = "123",
+    bundleId = "abc",
+    url = "http://example.com"
+  )
+  saveDeployment(
+    app_dir,
+    name = "test",
+    title = "",
+    username = "ron",
+    account = "ron",
+    server = "foo2",
+    hostUrl = "",
+    appId = "123",
+    bundleId = "abc",
+    url = "http://example.com"
+  )
+
+  expect_snapshot(error = TRUE, {
+    deploymentTarget(app_dir, appName = "test")
+    deploymentTarget(app_dir)
+  })
+})
+
+test_that("succeeds if there's a single existing deployment", {
+  mockr::local_mock(
+    accounts = fakeAccounts("ron", "bar"),
+    accountInfo = fakeAccountInfo(ron = list(
+      username = "foo",
+      server = "test"
+    ))
+  )
+
+  app_dir <- withr::local_tempdir()
+  file.create(file.path(app_dir, "app.R"))
+  saveDeployment(
+    app_dir,
+    name = "test",
+    title = "my title",
+    username = "ron",
+    account = "ron",
+    server = "bar",
+    hostUrl = "",
+    appId = "123",
+    bundleId = "abc",
+    url = "http://example.com"
+  )
+
+  target <- deploymentTarget(app_dir, appId = "123")
+  expect_equal(target$appId, "123")
+  expect_equal(target$appTitle, "my title")
+
+  target <- deploymentTarget(app_dir, appName = "test", appId = "123")
+  expect_equal(target$appId, "123")
+  # TODO(HW): why doesn't this equal my title?
+  # expect_equal(target$appTitle, "my title")
```

Comment:
I think this clause might be the cause (version before your PR):
https://github.com/rstudio/rsconnect/blob/ce49da6c57599137b8fe1c41c609ed4ad5e82e56/R/deployApp.R#L694-L699

We have a target deployment, but aren't using its stored data.

Reply:
Yeah, looks like I added a matching TODO here: https://github.com/rstudio/rsconnect/pull/662/files#diff-7284d883e7820ca83c32ae955a374305739a8841be7d9ec6d5860cbe2000e903R82

Reply:
It appears that we don't have any way to perform a title "update", so yes -- the stored title overrides any incoming user-provided title. The `connect.R` and `lucid.R` only define `createApplication` and don't offer a way to update `title`.

There are probably lurking inconsistencies, however. The `application` returned by `applicationForTarget` is conditionally fetched from the server, and I cannot see where its `title` (Connect-only) is written back to the deployment record.

Reply:
When I refactor this, I'll have a go at extracting out the "find existing application" in it's own logic to make it all more consistent.